geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/GenericProtocolServerConnection.java

```diff
@@ -77,7 +77,9 @@ protected void doOneMessage() {
       setClientDisconnectedException(e);
       logger.debug("Encountered EOF while processing message: {}", e);
     } catch (IOException | IncompatibleVersionException e) {
-      logger.warn(e);
+      if (!socket.isClosed()) { // GEODE-4300, IOException may be thrown thrown on EOF
```

Comment:
Do we need to put GEODE-4300 in the code since the commit comment also includes it?

Reply:
We were actually complemented on doing this when VMWare audited the GemFire source base, but I understand where you're coming from.